api/api/models/category.py

```diff
@@ -0,0 +1,15 @@
+from mongoengine import Document, ReferenceField, ListField, StringField
+from hardware import Hardware #noqa
+
+
+class Category(Document):
+    name = StringField(max_length=120)
+    products = ListField(ReferenceField('Hardware'))
+
+
+def get_all_categories():
+    return Category.objects.all()
+
+
+def get_category_by_id(id_):
```

Comment:
We might want to make Category.name unique so we can get by name. This will allow us to do `http://localhost/category/motherboard/products/{productId}`. Thoughts?